### PR TITLE
Fixes the caching problem of layout and adds a test.

### PR DESCRIPTION
**Pull request by sn1ckers**
more information at https://github.com/donpark/hbs/issues/114

### CODE REVIEW

**Comment by defunctzombie** [comment]
great fix, minor syntax fixup in test. Otherwise looks good.

File: test/3.x/no_layout_app.js [hunk 1]
```diff
@@ -27,16 +27,36 @@ app.use(express.static(__dirname + '/public'));
 
 app.get('/', function(req, res){
   res.render('no_layout', {
-    title: 'Express Handlebars Test',
+    title: 'Express Handlebars Test'
```
Comment (defunctzombie) [inline test/3.x/no_layout_app.js]: not something that needs changing

File: test/3.x/no_layout_app.js [hunk 2]
```diff
@@ -27,16 +27,36 @@ app.use(express.static(__dirname + '/public'));
...
   });
 });
 
 app.get('/with_layout', function(req, res){
   res.render('blank', {
-    layout: 'layout'
+    layout: 'layout',
+    title: 'Express Handlebars Test'
```
Comment (defunctzombie) [inline test/3.x/no_layout_app.js]: was this failing before?

File: test/4.x/no_layout_app.js [hunk 3]
```diff
@@ -27,16 +29,36 @@ app.use(express.static(__dirname + '/public'));
...
-    layout: 'layout'
+    layout: 'layout',
+    title: 'Express Handlebars Test'
   });
 });
 
+app.get('/layout_cache', function(req, res, next) {
+    var root = process.cwd(),
+        path = require('path'),
+        hbs = require('../../');
```
Comment (defunctzombie) [inline test/4.x/no_layout_app.js]: these requires should be at top level

File: test/3.x/no_layout_app.js [hunk 4]
```diff
@@ -27,16 +27,36 @@ app.use(express.static(__dirname + '/public'));
...
   });
 });
 
+app.get('/layout_cache', function(req, res, next) {
+    var root = process.cwd(),
+        path = require('path'),
+        hbs = require('../../');
```
Comment (defunctzombie) [inline test/3.x/no_layout_app.js]: these requires should be at top level and follow existing style

File: test/3.x/no_layout_app.js (hunk 2 above)
Comment (sn1ckers) [inline test/3.x/no_layout_app.js]: I added this test and then they where failing on the title and blank.hbs hade a linebreak that I removed, then they worked. Im not sure if i put the test in the right test suite but it seemed like the most equivalent.

**Comment by sn1ckers** [comment]
I also noticed that the existing test **response locals cached** is not testing the cache it just use same url twice and if it is cached or not is not tested.

with a closer look I maybe should have put my test in the **view engine** and maybe remove **no_layout_app**?

**Comment by defunctzombie** [comment]
Either way works. Could be fine to put the test into view engine if that makes sense to you.

Please just squash the style changes commit back into the one before it.

Test placement is up to you, looks good otherwise.

**Comment by sn1ckers** [comment]
I think I finally got the squash right.

**Comment by defunctzombie** [comment]
There should be just two commits (or one if it makes your life easier). These other merge master or whatnot are not necessary.

**Comment by sn1ckers** [comment]
Im kind of new to git, I backed my master so now it should be alright.

**Comment by defunctzombie** [comment]
:+1:

**Pull request merged**